### ISSUE TITLE
Repeated Oh, Oh, on low volume

### ISSUE THREAD

**Issue by ghost** [futo-org/voice-input#19]
![Screenshot_20231017_131623_AnySoftKeyboard.png](https://github.com/futo-org/voice-input/assets/75695425/1dba1b91-1ac9-4eaa-aca1-9ce5a441fc20)
 Oh, oh, oh :(

**Comment by rktfier** [comment]
Sorry about the above, the app seems to do this on low voice surroundings or when i don't speak at my full volume to it. I'll be happy to add details once i get back to my PC.

**Comment by rktfier** [comment]
So this has been happening when i'm trying to whisper into the microphone (not really, but speak really really softly) for a small bit. I don't mind if it told me something of "could not detect voice" but it's pretty weird how it just repeats the same word again and again, and i have no idea what causes it to be. This happened inside of a library, or when just in my room trying not to disturb anyone sleeping at 3am.

**Comment by abb128** [comment]
Whisper is basically a transformer language model, so it can have weird behavior at times. It can get stuck in a loop of repeating something like this if the audio is unclear. The official Whisper Python implementation tries to detect this by compressing the text with gzip and treating it as failed if the size difference ratio is beyond a certain threshold (`compression_ratio_threshold`, defaults to 2.4). It has thresholds for other things as well to control the model a bit better in these failure cases. I may look into implementing these at some point.

**Comment by rktfier** [comment]
Yeah i found out the hard way it doesn't do this only with "Oh, Oh" and it will just repeat a sentence over and over.

**Comment by David-Prock** [comment]
> ![Screenshot_20231017_131623_AnySoftKeyboard.png](https://github.com/futo-org/voice-input/assets/75695425/1dba1b91-1ac9-4eaa-aca1-9ce5a441fc20)
>  Oh, oh, oh :(
> 

![Screenshot_20231211-153115.png](https://github.com/futo-org/voice-input/assets/127499434/486a065c-bd30-416c-ba2e-58cf09de7627)

**Comment by David-Prock** [comment]
I juts had a ha, ha, ha, laughing to deaf loop
